transports/small-webrtc-transport/src/smallWebRTCTransport.ts

```diff
@@ -0,0 +1,640 @@
+        } else if (event.type == "video") {
+          logger.info("SmallWebRTCMediaManager replacing video track");
+          await this.getVideoTransceiver().sender.replaceTrack(event.track);
+        }
+      },
+      (event) => logger.debug("SmallWebRTCMediaManager Track stopped:", event),
+    );
+  }
+
+  public initialize(
+    options: RTVIClientOptions,
+    messageHandler: (ev: RTVIMessage) => void,
+  ): void {
+    this._options = options;
+    this._callbacks = options.callbacks ?? {};
+    this._onMessage = messageHandler;
+    this.mediaManager.setRTVIOptions(options);
+
+    if (this._options.params.config?.length || 0 > 0) {
+      let config = this._options.params.config![0];
+      if (
+        config.service == SmallWebRTCTransport.SERVICE_NAME &&
+        config.options.length > 0
+      ) {
+        config.options.forEach((option) => {
+          if (option.name == "audioCodec") {
+            this.audioCodec = option.value as string;
+          } else if (option.name == "videoCodec") {
+            this.videoCodec = option.value as string;
+          }
+        });
+      }
+    }
+
+    this.state = "disconnected";
+    logger.debug("[RTVI Transport] Initialized");
+  }
+
+  async initDevices() {
+    this.state = "initializing";
+    await this.mediaManager.initialize();
+    this.state = "initialized";
+  }
+
+  setAudioCodec(audioCodec: string | null): void {
+    this.audioCodec = audioCodec;
+  }
+
+  setVideoCodec(videoCodec: string | null): void {
+    this.videoCodec = videoCodec;
+  }
+
+  async connect(
+    authBundle: unknown,
+    abortController: AbortController,
+  ): Promise<void> {
+    if (abortController.signal.aborted) return;
+
+    this.state = "connecting";
+
+    await this.mediaManager.connect();
+
+    await this.startNewPeerConnection();
+
+    if (abortController.signal.aborted) return;
+
+    // Wait until we are actually connected and the data channel is ready
+    await new Promise<void>((resolve, reject) => {
+      this._connectResolved = resolve;
+      this._connectFailed = reject;
+    });
+
+    this.state = "connected";
+    this._callbacks.onConnected?.();
+  }
+
+  sendReadyMessage() {
+    // Sending message that the client is ready, just for testing
+    //this.dc?.send(JSON.stringify({id: 'clientReady', label: 'rtvi-ai', type:'client-ready'}))
+    this.sendMessage(RTVIMessage.clientReady());
+  }
+
+  sendMessage(message: RTVIMessage) {
+    if (!this.dc || this.dc.readyState !== "open") {
+      logger.warn(`Datachannel is not ready. Message not sent: ${message}`);
+      return;
+    }
+    this.dc?.send(JSON.stringify(message));
+  }
+
+  async disconnect(): Promise<void> {
+    this.state = "disconnecting";
+    await this.stop();
+    this.state = "disconnected";
+  }
+
+  private createPeerConnection(): RTCPeerConnection {
+    const config: RTCConfiguration = {
+      iceServers: [{ urls: this._iceServers }],
```

Comment:
instead of `this._iceServers` being just a string array for the urls, should we make it support providing the full config? This is what we do in daily-js

Reply:
Considering all the [other possible options](https://udn.realityripple.com/docs/Web/API/RTCConfiguration), I think I am ok keeping it as simple as just providing the ice_servers. We can always revisit that if needed.